### ISSUE TITLE
About 5-shot train and test

### ISSUE THREAD

**Issue by bowen099** [kosukesakurai1/VLP-SAM#2]
Hello, I would like to perform 5-shot training and testing with VLP-SAM. Could you please advise me on what changes I need to make? I would appreciate your general suggestions. Thank you for your guidance!

**Comment by kosukesakurai1** [comment]
To implement 5-shot training, you may consider modifying the pipeline to concatenate 5-shot 'protos' output from VLP encoder ('model').

**Comment by bowen099** [comment]
Thank you for your suggestion!
Additionally, I noticed that the SAM model is always frozen in the text. In the code, I found that SAM is set to evaluation mode. Does this mean that the parameters of SAM are not being updated?

sam_vith_weightpath = os.path.join(self.weightpath, 'SAM/sam_vit_h_4b8939.pth')
self.sam_model = sam_model_registry['vit_h'](sam_vith_weightpath)
self.sam_model.eval()

**Comment by kosukesakurai1** [comment]
Yes!
SAM model is frozen during training, and its parameters are not updated.

**Comment by bowen099** [comment]
![Image](https://github.com/user-attachments/assets/c090357b-f196-4468-8752-522d0f93c2ed)

![Image](https://github.com/user-attachments/assets/c2467c65-515f-43f2-a67c-518f3fd20174)

Thank you for your reply. In the Pascal 1-shot experiment, I found that the model achieved its best performance at the 69th epoch (best model). The corresponding validation results at that point are shown in the figure. May I ask if the F-0:76.94 mentioned in the paper refers to the validation result at epoch 69? Why higher than you？

**Comment by kosukesakurai1** [comment]
The performance can slightly vary depending on the random seed and training environment, so please consider the results in our table as a general reference.

Your results look very close in our mIoU, so I’d say your training was successful.

**Comment by foreverzzx** [comment]
@bowen099 @kosukesakurai1 hey, have you tried that increasing the number of referenced images? Will it improve the performance of segmentation?
